unittests/test_pipeline.py

```diff
@@ -498,6 +529,32 @@ def check_executable(self):
         test = MyTest()
         _run(test, self.partition, self.prgenv)
 
+    def test_run_hooks_in_run_only_test(self):
+        @fixtures.custom_prefix('unittests/resources/checks')
+        class MyTest(rfm.RunOnlyRegressionTest):
+            def __init__(self):
+                self.executable = './hello.sh'
+                self.executable_opts = ['Hello, World!']
+                self.local = True
+                self.valid_prog_environs = ['*']
+                self.valid_systems = ['*']
+                self.sanity_patterns = sn.assert_found(
+                    r'Hello, World\!', self.stdout)
+
+            @rfm.run_before('run')
+            def setflags(self):
+                self.post_run = ['echo hello > greetings.txt']
```

Comment:
This should be enough:

```suggestion
                self.post_run = ['touch greetings.txt']
```